IdentityCore/src/cache/token/MSIDCredentialCacheItem.m

```diff
@@ -316,13 +323,11 @@ - (BOOL)matchesWithRealm:(nullable NSString *)realm
 
     if (clientIDMatchingOptions == MSIDSuperSet)
     {
-        if ((clientId && [self.clientId.msidNormalizedString isEqualToString:clientId.msidNormalizedString])
-            || (familyId && [self.familyId.msidNormalizedString isEqualToString:familyId.msidNormalizedString]))
+        if (!(clientId && [self.clientId.msidNormalizedString isEqualToString:clientId.msidNormalizedString])
```

Comment:
Why this one got updated? Doesn't seem to be related to the changes in this PR.

Reply:
Previously, this if was an OR that returned YES (exiting out) if either condition is true. Given that I added my new requested_claims comparison check at the end of the matcher, after this statement, I inverted the IF to fail out if both conditions are untrue, and to continue evaluation (down to my new check) if the condition is satisfied. 

If I were to revert this change, I'd either have to duplicate my requested_claims check and put a 2nd copy inside this if statement, or I'd have to move it up in the matcher, and IIRC split it into 2 separate if statements as opposed to the single if I put at the end. 

Happy to change it though if you'd rather I don't touch this :)

Reply:
Previously, this if was an OR that returned YES (exiting out) if either condition is true. - I think this is by design :) This is needed for some account removal operations in MSAL code where we want to remove all tokens for the same clientId or familyId if my memory serves me well.

Reply:
Fixed in latest commit :)